greymatter-io/vscode-greymatter
# add ci

Resolves #3 
Resolves #2 

- Added a circleci config with 2 jobs: build and publish. Build is run on every commit. Publish is run on version tags, and pushes a tar to nexus with the license, readme, version, and vsix package. 
- Added a VERSION file and script to create it (from the dashboard)
- Added a license
- The readme is used as the user-facing document in the vscode marketplace. I updated this so it's more appropriate, and moved dev stuff to a CONTRIBUTING.md file.
- Updated package.json with metadata used in the vscode marketplace


To test:

- Read through README.md
- [Look through the completed circle job that I ran to test the workflows](https://app.circleci.com/pipelines/github/greymatter-io/vscode-greymatter/28/workflows/4fbd468e-ef05-440e-8952-921fdd1ea93c) (v0.0.2 - I can unpublish this after review, I just couldn't test pushing to the marketplace without using a real version.) 
   - See that the 0.0.2 tar was successfully [pushed to nexus](https://nexus.greymatter.io/#browse/browse:raw:development%2Fvscode-greymatter%2Fvscode-greymatter-0.0.2.tar.gz)
   - See that 0.0.2 was successfully [pushed to vscode](https://marketplace.visualstudio.com/items?itemName=greymatter.greymatter)
